openslides_backend/action/actions/projector/delete.py

```diff
@@ -0,0 +1,37 @@
+from typing import Any, Dict
+
+from ....models.models import Projector
+from ....shared.exceptions import ActionException
+from ....shared.patterns import FullQualifiedId
+from ...generics.delete import DeleteAction
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ..meeting.replace_projector_id import MeetingReplaceProjectorId
+
+
+@register_action("projector.delete")
+class ProjectorDelete(DeleteAction):
+    """
+    Action to delete a projector.
+    """
+
+    model = Projector()
+    schema = DefaultSchema(Projector()).get_delete_schema()
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        projector = self.datastore.get(
+            FullQualifiedId(self.model.collection, instance["id"]),
+            ["used_as_reference_projector_meeting_id", "meeting_id"],
+        )
+        if projector.get("used_as_reference_projector_meeting_id"):
+            raise ActionException(
+                "A used as reference projector is not allowed to delete."
+            )
+        instance["remember_meeting_id"] = projector.get("meeting_id")
```

Comment:
can you save this as a class member?

Reply:
class member is no good idea, because that would impact all instances of class.
class-instance member is less risky, but could fail in actions with more projectors (and different meeting's to create) to act with.
To be secure, we could use an instance-dict-object. But in IMO the instance-dict is the simplest and most intuitive method. But ugly! ;-)